src/app/state/reducers/datavault.ts

```diff
@@ -0,0 +1,38 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+export interface DataVaultKey {
+  key: string,
+  content: string[]
+}
```

Comment:
I think is better model if we do a map for key->contents. That will also improve operations declarative

Reply:
Yes, I thought this too especially after the comment in the tokens PR a few days ago. So I looked into it and found that redux doesn't recommend having map in the redux state. To put maps in the state, you have to turn off a flag in the config store which seemed like a smell.

From the documentation: 

> Avoid putting non-serializable values such as Promises, Symbols, Maps/Sets, functions, or class instances into the Redux store state or dispatched actions

https://redux.js.org/style-guide/style-guide#do-not-put-non-serializable-values-in-state-or-actions

Reply:
I didn’t mean using `Map`. We can implement it with an object.

Reply:
I looked at this again and I figured it out! It was the typings that I had the most trouble with :-)

I have implemented this in the last commit when I also implemented the beta2 version of the ipfs pinner.